KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/AuthenticationModule.java

```diff
@@ -75,20 +76,18 @@ private KalturaLiveEntry onClientConnect(WMSProperties properties, HashMap<Strin
 
         KalturaEntryServerNodeType serverIndex = KalturaEntryServerNodeType.get(requestParams.get(Constants.REQUEST_PROPERTY_SERVER_INDEX));
         KalturaLiveEntry liveEntry = KalturaAPI.getKalturaAPI().authenticate(entryId, partnerId, token, serverIndex);
-        synchronized(properties) {
-            properties.setProperty(Constants.CLIENT_PROPERTY_SERVER_INDEX, requestParams.get(Constants.REQUEST_PROPERTY_SERVER_INDEX));
-            properties.setProperty(Constants.CLIENT_PROPERTY_KALTURA_LIVE_ENTRY, liveEntry);
-        }
+        KalturaEntryDataPersistence.setProperty(entryId, Constants.CLIENT_PROPERTY_KALTURA_LIVE_ENTRY, liveEntry);
         logger.info("Entry added [" + entryId + "]");
 
         return liveEntry;
     }
 
-
     public void onDisconnect(IClient client) {
         try{
-            KalturaLiveEntry liveEntry = Utils.getKalturaLiveEntry(client);
-            logger.info("Entry removed [" + liveEntry.id + "]");
+            // TODO: entryID returns null -> Check with Lilach!
```

Comment:
pls set in client properties the entry id during onConnect(). Then, when onDisconnect() is called, get it from the client properties.